modal/secret.py

```diff
@@ -54,8 +55,40 @@ async def _load(resolver: Resolver, existing_object_id: Optional[str]) -> _Secre
             resp = await resolver.client.stub.SecretCreate(req)
             return _SecretHandle._from_id(resp.secret_id, resolver.client, None)
 
-        rep = f"Secret([{', '.join(env_dict.keys())}])"
-        super().__init__(_load, rep)
+        rep = f"Secret.from_dict([{', '.join(env_dict.keys())}])"
+        return _Secret._from_loader(_load, rep)
+
+    def __init__(self, env_dict: Dict[str, str]):
+        """`Secret({...})` is deprecated. Please use `Secret.from_dict({...})` instead."""
+        deprecation_warning(date(2023, 5, 1), self.__init__.__doc__)
+        obj = _Secret.from_dict(env_dict)
+        self._init_from_other(obj)
+
+    @staticmethod
+    def from_dotenv():
+        async def _load(resolver: Resolver, existing_object_id: Optional[str]) -> _SecretHandle:
```

Comment:
Should we throw an explicit error here if we are on <py3.8? (given the dependency condition), or is the import error meant to indicate that issue?